cartography/intel/gcp/crm/orgs.py

```diff
@@ -3,30 +3,38 @@
 from typing import List
 
 import neo4j
-from googleapiclient.discovery import HttpError
-from googleapiclient.discovery import Resource
+from google.cloud import resourcemanager_v3
 
-from cartography.util import run_cleanup_job
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.models.gcp.crm.organizations import GCPOrganizationSchema
 from cartography.util import timeit
 
 logger = logging.getLogger(__name__)
 
 
 @timeit
-def get_gcp_organizations(crm_v1: Resource) -> List[Dict]:
+def get_gcp_organizations() -> List[Dict]:
     """
-    Return list of GCP organizations that the crm_v1 resource object has permissions to access.
-    Returns empty list if we are unable to enumerate organizations for any reason.
-    :param crm_v1: The Compute Resource Manager v1 resource object created by `googleapiclient.discovery.build()`.
-    :return: List of GCP Organizations.
+    Return list of GCP organizations that the authenticated principal can access using the high-level client.
+    Returns empty list on error.
+    :return: List of org dicts with keys: name, displayName, lifecycleState.
     """
+    client = resourcemanager_v3.OrganizationsClient()
     try:
-        req = crm_v1.organizations().search(body={})
-        res = req.execute()
-        return res.get("organizations", [])
-    except HttpError as e:
+        orgs = []
+        for org in client.search_organizations():
+            orgs.append(
+                {
+                    "name": org.name,
+                    "displayName": org.display_name,
+                    "lifecycleState": org.state.name,
+                }
+            )
+        return orgs
+    except Exception as e:
```

Comment:
I wonder if this module is mature enough to get rid of the try+except clause so that we can crash loudly instead of burying and continuing (potentially with cleanup jobs deleting old data). Wdyt?